File: hello.py
from android import Activity, LinearLayout, TextView, Toast

class MainActivity(Activity):
    def mainLayout(self):
        layout = LinearLayout(self)
        layout.orientation = LinearLayout.VERTICAL
        
        text = TextView(self)
        text.text = "Hello, world!"
        layout.addView(text)
        
        return layout

    def onCreate(self, savedInstanceState):
        super(MainActivity, self).onCreate(savedInstanceState)
        setContentView(mainLayout(self))
        
        toast = Toast.makeText("Goodbye, world!")
        toast.show()

File: hello.java
package com.example.helloworld;

import android.app.Activity;
import android.os.Bundle;
import android.widget.LinearLayout;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends Activity {
	
	public static LinearLayout mainLayout(Activity activity){
		LinearLayout layout = new LinearLayout(activity);
		layout.setOrientation(LinearLayout.VERTICAL);
		
		TextView text = new TextView(activity);
		text.setText("Hello, world!");
		layout.addView(text);
		
		return layout;
	}

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(mainLayout(this));
		
		Toast toast = Toast.makeText(getApplicationContext(), "Goodbye, world!", Toast.LENGTH_LONG);
		toast.show();
	}
}